apps/ai-image-tagging/frontend/src/components/AITagView/AITagView.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { CheckboxField, TextInput, Pill, Button } from '@contentful/forma-36-react-components';
+
+import { styles } from './styles';
+
+async function callAPI(url) {
+  const data = await fetch(`http://localhot:3000/tags/${url}`).then(res => res.json());
+  return data.tags;
+}
+
+export class AITagView extends React.Component {
+
+  constructor(props){
+    super(props);
+
+    this.state = {
+      value: "",
+      checked: false,
+      tags: props.entries.imageTags.getValue() || [],
+      overwrite: true,
+      isMissingImage: !props.entries.image.getValue(),
+      isFetchingTags: false
+    }
+  }
+
+  componentDidMount() {
+    this.props.entries.image.onValueChanged(() => {
+      this.setState(() => ({
+        isMissingImage: !this.props.entries.image.getValue()
+      }))
+    });
+  }
+
+  toggleOverwrite = () => {
+    this.setState(state => ({
+      overwrite: !state.overwrite
+    }));
+  }
+
+  addTag = async (e) => {
+    if (e.key !== "Enter") { return; }
+
+    const newTags = [e.target.value, ...this.state.tags];
+    try {
```

Comment:
Why `try/catch`

Reply:
`async/await` pattern for doing something with errors. Not sure how to handle them yet but at least this way I dont forget

Reply:
yeah, but now you just swallow them so you won't notice when they happen so it's more likely to forget